sources/bing_webmaster/__init__.py

```diff
@@ -0,0 +1,92 @@
+"""
+A source loading history of organic search traffic from Bing Webmaster API
+See documentation: https://learn.microsoft.com/en-us/dotnet/api/microsoft.bing.webmaster.api.interfaces.iwebmasterapi?view=bing-webmaster-dotnet
+"""
+
+import time
+from typing import Iterable, Iterator, List, Sequence
+
+import dlt
+from dlt.common.typing import DictStrAny, DictStrStr
+from dlt.sources import DltResource
+
+from .helpers import get_stats_with_retry, parse_response
+
+
+@dlt.source(name="bing_webmaster")
+def source(
+    site_urls: List[str] = None, site_url_pages: Iterable[DictStrStr] = None
+) -> Sequence[DltResource]:
+    """
+    A dlt source for the Bing Webmaster api.
+    It groups resources for the APIs which return organic search traffic statistics
+    Args:
+        site_urls: List[str]: A list of site_urls, e.g, ["dlthub.com", "dlthub.de"]. Use this if you need the weekly traffic per site_url and page
+        site_url_pages: Iterable[DictStrStr]: A list of pairs of site_url and page. Use this if you need the weekly traffic per site_url, page, and query
+    Returns:
+        Sequence[DltResource]: A sequence of resources that can be selected from including page_stats and page_query_stats.
+    """
+    return (
+        page_stats(site_urls),
+        page_query_stats(site_url_pages),
+    )
+
+
+@dlt.resource(
+    write_disposition="merge",
+    merge_key=("date", "page", "site_url"),
+    primary_key=("date", "page", "site_url"),
+    table_name="bing_page_stats",
+)
+def page_stats(
+    site_urls: List[str], bing_webmaster_api_key: str = dlt.secrets.value
```

Comment:
Wrt `bing_webmaster_api_key` should we exit/raise exception if key is not set?

Reply:
Good idea! Do you have an exception in mind that we could use? `ValueError`?

I couldn't find any example yet in the verified sources.

Reply:
I think it is fine to not raise any exception since requests will fail with HTTP 400 right?

Reply:
Yes, I removed my check code because even without it we have the following behaviour:


### Without any key set
Exception:
```
dlt.common.configuration.exceptions.ConfigFieldMissingException: Following fields are missing: ['api_key'] in configuration with spec PageStatsConfiguration
	for field "api_key" config providers and keys were tried in following order:
		In Environment Variables key BING_WEBMASTER_PAGE_STATS__SOURCES__BING_WEBMASTER__PAGE_STATS__API_KEY was not found.
		In Environment Variables key BING_WEBMASTER_PAGE_STATS__SOURCES__BING_WEBMASTER__API_KEY was not found.
		In Environment Variables key BING_WEBMASTER_PAGE_STATS__SOURCES__API_KEY was not found.
		In Environment Variables key BING_WEBMASTER_PAGE_STATS__API_KEY was not found.
		In secrets.toml key bing_webmaster_page_stats.sources.bing_webmaster.page_stats.api_key was not found.
		In secrets.toml key bing_webmaster_page_stats.sources.bing_webmaster.api_key was not found.
		In secrets.toml key bing_webmaster_page_stats.sources.api_key was not found.
		In secrets.toml key bing_webmaster_page_stats.api_key was not found.
		In Environment Variables key SOURCES__BING_WEBMASTER__PAGE_STATS__API_KEY was not found.
		In Environment Variables key SOURCES__BING_WEBMASTER__API_KEY was not found.
		In Environment Variables key SOURCES__API_KEY was not found.
		In Environment Variables key API_KEY was not found.
		In secrets.toml key sources.bing_webmaster.page_stats.api_key was not found.
		In secrets.toml key sources.bing_webmaster.api_key was not found.
		In secrets.toml key sources.api_key was not found.
		In secrets.toml key api_key was not found.
Please refer to https://dlthub.com/docs/general-usage/credentials for more information
```

### With a wrong key
Exception:
```
requests.exceptions.HTTPError: 400 Client Error: Error for url: https://ssl.bing.com/webmaster/api.svc/json/GetPageStats?siteUrl=satellite.me&apikey=my-wrong-key
```